mmsoap/client.py

```diff
@@ -72,58 +55,77 @@ def create(self, object_name):
         return self.client.factory.create(object_name)
 
     def set_userId(self, userId):
+        """ Set user id on the authentication element. """
         self.authentication.userId = userId
 
     def set_password(self, password):
+        """ Set password on the authentication element. """
         self.authentication.password = password
 
     def check_user(self):
-        return self.client.service.checkUser(self.authentication)\
-            .accountDetails
-
-    def send_messages(self, recipients, content, send_mode='normal'):
         """
-        Send a message to `recipients`.
+        Check user account details.
 
-        Arguments:
-        recipients -- iterable of recipient numbers
-        content -- message content
+        :return: The account details including used and remaining credit limits.
+        """
+        return self.client.service.checkUser(self.authentication).accountDetails
 
-        Keyword arguments:
-        send_mode -- used for testing. One of:
-            * normal (default) -- send as normal
-            * dropAll -- drop (not send) the requested messages, return
-              random errors and successes
-            * dropAllWithErrors -- drop the requested messages,
-              return errors
-            * dropAllWithSuccess -- drop the requested messages,
-              return success
+    def send_messages(self, recipients, content, send_mode='normal', scheduled=None, report=None, id=None):
         """
-        recipients_type = self.create("RecipientsType")
+        Send a single message to `recipients`.
+
+        :param recipients: Iterable of recipient numbers
+        :param content: Message content
+        :param send_mode: Used for testing. Should be one of:
+                            * normal (default) -- send as normal.
+                            * dropAll -- drop (not send) the requested messages, return a mix of errors and successes.
+                            * dropAllWithErrors -- drop the requested messages, return all as errors.
+                            * dropAllWithSuccess -- drop the requested messages, return all as success.
+        :param scheduled: Scheduled date/time of the message in UTC format.
+        :param report: Request a delivery report for this message.
+
+        :return: Response containing count of sent, scheduled, failed and any associated error code types.
+        """
+        recipients_type = self.create('RecipientsType')
 
         for recipient in recipients:
-            this_recipient = self.create("RecipientType")
+            this_recipient = self.create('RecipientType')
```

Comment:
This is not really something we can just do now and do properly. the recipients parameter on this method should be turned into an object or a dict since the uid relates to each recipient. This would introduce a breaking change. The other alternative is to add a new param like an array of id's and just hope that each of these matches the index of the recipients but this is ugly and dangerous. Needs to be discussed further and this was just a 1 point story to add example code.